pkg/grizzly/parsing.go

```diff
@@ -18,13 +20,50 @@ import (
 )
 
 func Parse(registry Registry, opts GrizzlyOpts) (Resources, error) {
-	if strings.HasSuffix(*opts.ResourceFile, ".yaml") ||
-		strings.HasSuffix(*opts.ResourceFile, ".yml") {
-		return ParseYAML(registry, *opts.ResourceFile, opts)
-	} else if strings.HasSuffix(*opts.ResourceFile, ".jsonnet") ||
-		strings.HasSuffix(*opts.ResourceFile, ".libsonnet") ||
-		strings.HasSuffix(*opts.ResourceFile, ".json") {
-		return ParseJsonnet(registry, *opts.ResourceFile, opts)
+	if opts.ResourceFile != nil && opts.Directory != nil {
+		return nil, fmt.Errorf("Only specify a directory or a resource file, not both")
+	}
+	if opts.ResourceFile != nil {
+		return ParseFile(registry, opts, *opts.ResourceFile)
+	}
+	if opts.Directory != nil {
+		resources := Resources{}
```

Comment:
Super Nit: no need for a struct literal if you don't wanna set values
```suggestion
		var resources Resources
```